### PR TITLE
Add imu_sensor semantic component

### PR DESCRIPTION
**Pull request by v-lopez**
Ripoff from https://github.com/ros-controls/ros2_control/pull/370

### CODE REVIEW

File: controller_interface/include/semantic_components/imu_sensor.hpp [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+#ifndef SEMANTIC_COMPONENTS__IMU_SENSOR_HPP_
+#define SEMANTIC_COMPONENTS__IMU_SENSOR_HPP_
+
+#include <limits>
+#include <string>
+#include <vector>
+
+#include "hardware_interface/loaned_state_interface.hpp"
+#include "semantic_components/semantic_component_interface.hpp"
+#include "sensor_msgs/msg/imu.hpp"
```
Comment (bmagyar) [inline controller_interface/include/semantic_components/imu_sensor.hpp:24]: Thy dependencies need to be updateth

File: controller_interface/test/test_imu_sensor.hpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "gmock/gmock.h"
+
+#include "semantic_components/imu_sensor.hpp"
+
+// implementing and friending so we can access member varibles
```
Comment (bmagyar) [inline controller_interface/test/test_imu_sensor.hpp:30]:
```suggestion
// implementing and friending so we can access member variables 
```

File: controller_interface/test/test_imu_sensor.hpp [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+class TestableIMUSensor : public semantic_components::
+  IMUSensor
+{
+  FRIEND_TEST(IMUSensorTest, validate_all);
+
+public:
+  // Use generation of interface names
```
Comment (bmagyar) [inline controller_interface/test/test_imu_sensor.hpp:37]: Would be good to sprinkle this with a virtual destructor as per #428

File: controller_interface/include/semantic_components/imu_sensor.hpp (hunk 1 above)
Comment (v-lopez) [reply controller_interface/include/semantic_components/imu_sensor.hpp:23]: Is this https://github.com/ros-controls/ros2_control/pull/429/commits/b8ff6c6b7839f7cc99ce26693bd6a7cf33fdb804#diff-d3a44eabaa35ebf41dcc01ca13650700fbc1dda7dfa18abf3defc3e465a6f85eL22 that evil? If not please make me worthy of the secrets to fight such evil!

Comment (v-lopez) [reply controller_interface/include/semantic_components/imu_sensor.hpp:23]: nvm, package.xml dependencies. Fixed

Comment (bmagyar) [reply controller_interface/include/semantic_components/imu_sensor.hpp:23]: you are worthy now!

**Review by bmagyar — APPROVED** [review]
LGTM

**Review by jordan-palacios — APPROVED** [review]
GJ

**Pull request merged**